stream/src/main/java/org/apache/kafka/stream/KStream.java

```diff
@@ -150,7 +134,7 @@
     /**
      * Processes all elements in this stream by applying a processor.
      *
-     * @param processor the instance of Processor
+     * @param processorClass the class of Processor
      */
-    <K1, V1> KStream<K1, V1> process(KafkaProcessor<K, V, K1, V1> processor);
+    <K1, V1> KStream<K1, V1> process(Class<? extends KafkaProcessor<K, V, K1, V1>> processorClass, PConfig config);
```

Comment:
If we are taking this approach, i.e. giving a class object and a config object instead of an instance, we should clearly document that Predicate, KeyValueMapper, ValueMapper, etc. should not have any state and/or replace them with Java8 lambda expressions.

Alternatively consider removing them altogether by removing filer, map, etc. It is awkward to see philosophically different two approaches in the same interface.